tests/integration_tests/util.py

```diff
@@ -42,6 +45,130 @@ def verify_ordered_items_in_text(to_verify: list, text: str):
         index = matched.start()
 
 
+def format_found(header: str, items: list) -> str:
+    """Helper function to format assertion message"""
+
+    # do nothing, allows this formatter to be "stackable"
+    if not items:
+        return ""
+
+    # if only one error put the header and the error message on a single line
+    if 1 == len(items):
+        return f"\n{header}: {items.pop(0)}"
+
+    # otherwise make a list after header
+    else:
+        return f"\n{header}:\n\t- " + "\n\t- ".join(items)
+
+
+def verify_clean_boot(
```

Comment:
> At the very least I think the new function should call verify_clean_log for now.

I think that at the moment, verify_clean_log just won't work unless you pass matching expected failures. This is the result of the "Found unexpected errors/warnings" code paths - it just won't work unless you've passed everything that you expect to see in the `ignored_warnings` and `ignored_errors` keys. I don't think that calling `verify_clean_log()` inside of `verify_clean_boot()` will fix that - in fact we can't because we _require_ failures in some of our tests, but I'm happy to call that manually as well in the integration tests that we expect a clean log in.

One idea that I envision for this helper is a matrix of platforms and series that allows us to specify exactly when an expected warning message that will be seen. This would be more precise than the current "ignore this pattern everywhere" paradigm that `verify_clean_log()` uses. That said, I'm not sure that I want to include that refactor in this PR, which has already grown sizeable.